test/org/sagebionetworks/bridge/hibernate/HibernateSharedModuleMetadataTest.java

```diff
@@ -0,0 +1,156 @@
+import org.sagebionetworks.bridge.json.BridgeObjectMapper;
+import org.sagebionetworks.bridge.models.sharedmodules.SharedModuleMetadata;
+import org.sagebionetworks.bridge.models.sharedmodules.SharedModuleType;
+
+public class HibernateSharedModuleMetadataTest {
+    private static final String MODULE_ID = "test-module";
+    private static final String MODULE_NAME = "Test Module";
+    private static final String MODULE_NOTES = "These are my notes for my module.";
+    private static final String MODULE_OS = "Android";
+    private static final Set<String> MODULE_TAGS = ImmutableSet.of("foo", "bar", "baz");
+    private static final int MODULE_VERSION = 3;
+    private static final String SCHEMA_ID = "test-schema";
+    private static final int SCHEMA_REV = 7;
+    private static final long SURVEY_CREATED_ON = 1337;
+    private static final String SURVEY_GUID = "test-survey-guid";
+
+    @Test
+    public void getModuleTypeSchema() {
+        SharedModuleMetadata metadata = SharedModuleMetadata.create();
+        metadata.setSchemaId(SCHEMA_ID);
+        metadata.setSchemaRevision(SCHEMA_REV);
+        assertEquals(SharedModuleType.SCHEMA, metadata.getModuleType());
+    }
+
+    @Test
+    public void getModuleTypeSurvey() {
+        SharedModuleMetadata metadata = SharedModuleMetadata.create();
+        metadata.setSurveyCreatedOn(SURVEY_CREATED_ON);
+        metadata.setSurveyGuid(SURVEY_GUID);
+        assertEquals(SharedModuleType.SURVEY, metadata.getModuleType());
+    }
+
+    @Test(expected = InvalidEntityException.class)
+    public void getModuleTypeNeither() {
+        SharedModuleMetadata.create().getModuleType();
+    }
+
+    @Test
+    public void tags() {
+        SharedModuleMetadata metadata = SharedModuleMetadata.create();
+
+        // tags starts out empty
+        assertTrue(metadata.getTags().isEmpty());
+
+        // set the tags
+        metadata.setTags(ImmutableSet.of("foo", "bar", "baz"));
+        assertEquals(ImmutableSet.of("foo", "bar", "baz"), metadata.getTags());
+
+        // Set tag set to null. It should come back as empty.
+        metadata.setTags(null);
+        assertTrue(metadata.getTags().isEmpty());
+    }
+
+    @Test
+    public void jsonSerializationWithOptionalFields() throws Exception {
+        // start with JSON
+        String jsonText = "{\n" +
+                "   \"id\":\"" + MODULE_ID + "\",\n" +
+                "   \"licenseRestricted\":true,\n" +
+                "   \"name\":\"" + MODULE_NAME + "\",\n" +
+                "   \"notes\":\"" + MODULE_NOTES + "\",\n" +
+                "   \"os\":\"" + MODULE_OS + "\",\n" +
+                "   \"published\":true,\n" +
+                "   \"schemaId\":\"" + SCHEMA_ID + "\",\n" +
+                "   \"schemaRevision\":" + SCHEMA_REV + ",\n" +
+                "   \"tags\":[\"foo\", \"bar\", \"baz\"],\n" +
+                "   \"version\":" + MODULE_VERSION + "\n" +
+                "}";
```

Comment:
This is verified later in the test when we convert it back into JSON. Or am I missing something?